src/SmoFacade/Database.cs

```diff
@@ -15,10 +15,35 @@ public class Database
     private readonly Microsoft.SqlServer.Management.Smo.Database _database;
     private readonly Server _server;
 
+    public Database(DatabaseConfig databaseConfig)
+    { 
+        string connectionString = databaseConfig.ConnectionString;
+        string dbName = databaseConfig.DatabaseName;
+
+        var sqlConnection = new System.Data.SqlClient.SqlConnection(connectionString);
+        var sqlServerConnection = new Microsoft.SqlServer.Management.Common.ServerConnection(sqlConnection);
+        var sqlSmoServer = new Microsoft.SqlServer.Management.Smo.Server(sqlServerConnection);
+        _database = new Microsoft.SqlServer.Management.Smo.Database(sqlSmoServer, dbName); 
+
+        _server = new Server(connectionString);
+        RefreshAndOpenSqlConnection();
+    }
+
     internal Database(Microsoft.SqlServer.Management.Smo.Database database, Server server)
     {
       _database = database;
       _server = server;
+      RefreshAndOpenSqlConnection();
+    }
+
+    ~Database()
+    {
+      _server.SqlConnection.Close();
+    }
+    private void RefreshAndOpenSqlConnection()
+    {
+      _server.SqlConnection.Close();

```

Comment:
newline between methods, please.
I see that calling `Open` on an already open connection throws, so I suppose this is fine